src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -62,9 +62,11 @@ public sealed class CosmosFhirDataStore : IFhirDataStore, IProvideCapability
 
         /// <summary>
         /// This is the maximum degree of parallelism for the SDK to use when querying physical partitions in parallel.
-        /// -1 means "System Decides", int.MaxValue sets the limit high enough that it shouldn't be limited.
+        /// -1 means "System Decides", any other value and the SDK will choose min(physical partitions, value).
+        /// During testing we noticed unexpected behavior when executing queries with max degree of parallelism values
+        /// higher than 1000 and hence we are setting a conservative value.for now.
         /// </summary>
-        internal const int MaxQueryConcurrency = int.MaxValue;
+        internal const int MaxQueryConcurrency = 500;
```

Comment:
Was this confirmed or observed? Also did -1 also timeout?

Reply:
Observed. Still waiting for confirmation/updates. Just wanted to get a build going in the meantime. -1 also failed to fetch records

Reply:
Lets create a bug in the CosmosDBv3 SDK and link it here also